src/osd/osd_types.h

```diff
@@ -1514,6 +1514,18 @@ struct pg_pool_t {
   bool is_stretch_pool() const {
     return peering_crush_bucket_count != 0;
   }
+  
+  std::optional<std::tuple<uint32_t,uint32_t,uint32_t,uint32_t>> maybe_peering_crush_data() const {
+    if (!is_stretch_pool()) {
+        return std::nullopt;
+    } else {
+        return std::make_optional(std::tuple<uint32_t,uint32_t,uint32_t,uint32_t>(peering_crush_bucket_count,
```

Comment:
Not sure whether it's gonna fly but maybe anonymous structs could help:

```cpp
struct pg_pool_t {
  struct {
    uint32_t peering_crush_bucket_count;
    uint32_t peering_crush_bucket_barrier;
    // and so on
  };
};

Reply:
but then i cant call it inside the optional.

Reply:
I see.